wdl/transforms/draft2/src/main/scala/wdl/transforms/draft2/wdlom2wom/WdlDraft2WomConditionalNodeMaker.scala

```diff
@@ -70,8 +49,8 @@ object If {
       case other => s"An if block must be given a boolean expression but instead got '${ifBlock.condition.toWomString}' (a ${other.toDisplayString})".invalidNel
     }
 
-    val innerGraphValidation: ErrorOr[Graph] = WdlGraphNode.buildWomGraph(
-      ifBlock,
+    // TODO Dan: can I avoid casting to Scope before using the implicit?
+    val innerGraphValidation: ErrorOr[Graph] = (ifBlock: Scope).toWomGraph(
```

Comment:
Flagging this question too

Reply:
Something's off about these WdlDraft2 - specific instances being bound to shared  types `Scope`, `If`, etc.  It wouldn't make sense to have multiple implementations bound to that common type because that would cause >1 associated implicit and that is a no-no (it won't compile).

Anyway as it is you should try simulacrum and see if it gives you some variance in types here.

I doubt it, because inheritance is not really a thing in the typeclass world AFAIK.  This maybe the best solution for now, especially since it works!

Reply:
Yeah, the packages are a bit arbitrary right now, this stuff could equally well (maybe *should* even) be in `draft2`. I'm intending the targets of these conversion implementations be draft-2 classes (backed up by shared code in mixins in `shared`. Look out for tidy-up part 2.